ui/packages/catalog-ui-search/src/main/webapp/component/query-src/query-src.view.js

```diff
@@ -16,12 +16,46 @@
 const Marionette = require('marionette')
 const _ = require('underscore')
 const $ = require('jquery')
-const template = require('./query-src.hbs')
 const CustomElements = require('js/CustomElements')
 const sources = require('component/singletons/sources-instance')
+import React from 'react'
 
 module.exports = Marionette.ItemView.extend({
-  template: template,
+  template(props) {
+    return (
+      <React.Fragment key={Math.random()}>
```

Comment:
❓ Seems strange to use this as the key. Why does this element need a key?

Reply:
It's definitely strange, but there's a reason we're doing it.  The functionality for this component ended up breaking when we updated our render for Marionette to delegate to React's render. 

Basically, beforehand, this component would get rendered on every model update.  In base Marionette, that means the entire DOM would be destroyed and rebuilt on each render.  However, our delegation to React changed that.  Now the rendering is smarter, doing only a diff.

However, this component depended on the base destructive behavior of Marionette. 

That behavior can be emulated in React by using a random key here on every render.  

In the future, this component will be converted entirely to React and this strange key business will go away.